healpix_alchemy/point.py

```diff
@@ -1,26 +1,59 @@
 """Spatial indexing for astronomical point coordinates."""
+import numpy as np
 from sqlalchemy.ext.declarative import declared_attr
 from sqlalchemy.ext.hybrid import hybrid_method, hybrid_property
 from sqlalchemy.schema import Column, Index
-from sqlalchemy.sql import and_
+from sqlalchemy.sql import and_, between
 from sqlalchemy.types import Float
 
-from .math import sind, cosd
 from .util import InheritTableArgs
 
 __all__ = ('Point',)
 
 
+def get_x_for_ra_dec(ra, dec):
+    return np.cos(np.deg2rad(ra)) * np.cos(np.deg2rad(dec))
+
+
+def get_y_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(ra)) * np.cos(np.deg2rad(dec))
+
+
+def get_z_for_ra_dec(ra, dec):
+    return np.sin(np.deg2rad(dec))
+
+
+def get_x_for_context(context):
```

Comment:
This and the two subsequent functions could potentially be simplified with a factory pattern:

```
def context_factory(variable_name):
    def get_variable_for_context(context):
        params = context.get_current_parameters()
        func = __module__.dict[f'get_{variable_name}_for_ra_dec']
        return func(params['ra'], params['dec'])
    return get_variable_for_context
```
```

Reply:
Sorry for late re-review

Reply:
I did something similar.